tests/boards/nrf52_bsim/egu/src/main.c

```diff
@@ -0,0 +1,505 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <string.h>
+
+#include <hal/nrf_egu.h>
+#include <hal/nrf_ppi.h>
+#include <hal/nrf_timer.h>
+
+#include <zephyr/ztest.h>
+
+#define TIMER_NO 2
+#define TIMER_INSTANCE NRFX_CONCAT_2(NRF_TIMER, TIMER_NO)
+#define TIMER_IRQ TIMER2_IRQn
+#define TIMER_INT NRF_TIMER_INT_COMPARE0_MASK
+#define TIMER_PRIORITY 5
+#define TIMER_DELAY_TICKS 100
+#define EGU_DELAY_USEC 200
+
+struct timer_isr_context {
+	unsigned int egu_channel;
+};
+
+void test_egu_channels_count(void)
+{
+	zassert_equal(16, nrf_egu_channel_count(NRF_EGU0), "NRF_EGU0 incorrect number of channels");
```

Comment:
Here I see duplicated code, did you consider make this part more generic?
My proposal (in pseudo code):
array  [NRF_EGU0 - NRF_EGU5]
and loop with asserts validation
for(test_egu_cnt=NRF_EGU0 to test_egu_cnt=NRF_EGU5)
{
zassert_equal(16, nrf_egu_channel_count(test_egu_cnt), "NRF_EGU%d incorrect number of channels", test_egu_cnt);
}